test/functional/feature_assumeutxo.py

```diff
@@ -17,8 +17,7 @@
 
 Interesting test cases could be loading an assumeutxo snapshot file with:
 
-- TODO: An invalid hash
-- TODO: Valid hash but invalid snapshot file (bad coin height or truncated file or
+- TODO: Valid hash but invalid snapshot file (bad coin height or
```

Comment:
Isn't the bad coin height test also already covered with the test where you made the fix?

Reply:
> I wonder how one could test a "valid" hash but invalid snapshot file

I would just generally understand this as taking a valid snapshot file and altering the content so that it still serializes correctly but results in a different hash. The way I understand it "valid hash" here means that the hash in the metadata of the file matches the hash in the chainparams.

> I guess one would have to add a hash of the invalid file to the source code.

That implies that the attacker is able to give the user a build that is not a release, no? (or we have made a huge mistake when adding the hash) If the attackers are able to do that they could do anything so it doesn't seem feasible to cover such a scenario in our tests.